nodejs/eks/cluster.ts

```diff
@@ -579,9 +596,68 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
         nodeSecurityGroupTags: args.nodeSecurityGroupTags,
         storageClasses: userStorageClasses,
         fargateProfile: fargateProfile,
+        oidcProvider: oidcProvider,
     };
 }
 
+// Adapted from https://git.io/JvGHB
+
+// Find the root CA cert in a chain of certs.
+function findCACertificate(certificate: any, parent?: any): any {
+    if (certificate?.issuerCertificate.fingerprint !== certificate.fingerprint) {
+      return findCACertificate(certificate.issuerCertificate, certificate);
+    }
+    return parent;
+}
+
+const THUMBPRINT_MAX_RETRIES: number = 12;
+const THUMBPRINT_SLEEP_INTERVAL: number = 5000; // In milliseconds.
+
+// Get the URL's issuing CA cert thumbprint.
+// This is used for OIDC provider configuration.
+async function getIssuerCAThumbprint(issuerUrl: pulumi.Output<string>, retriesLeft: number, interval: number): Promise<string> {
+    // For up to 60 seconds (12 retries @ 5000 ms), try to contact the issuer URL.
+    try {
+        return await new Promise((resolve, reject) => {
+            issuerUrl.apply(url => {
+                const options = {
+                    hostname: url,
+                    port: 443,
+                    rejectUnauthorized: false,
+                };
+                const req = https
+                    .get(options)
+                    .on("error", reject)
+                    .on("socket", socket => {
+                        socket.on("secureConnect", () => {
+                            const certificate = socket.getPeerCertificate(true);
+                            const fingerprint = findCACertificate(certificate).fingerprint;
+                            // Check if certificate is valid
+                            if (socket.authorized === false) {
+                                req.emit("error", new Error(socket.authorizationError));
+                                return req.abort();
+                            }
+                            resolve(
+                                fingerprint
+                                .split(":")
+                                .join("")
+                                .toLowerCase(),
+                            );
```

Comment:
Is there a primary reference for all this logic we could refer to?

Reply:
Added refs that detail the logic in the comments.